File: examples/javascript/email-verification.js
/**
 * InstaTools API - Email Verification Example
 * 
 * This example demonstrates how to verify email addresses for deliverability.
 */

const API_KEY = process.env.INSTATOOLS_API_KEY || 'YOUR_API_KEY';
const BASE_URL = 'https://api.instatools.com';

/**
 * Verify a single email address
 */
async function verifyEmail(email) {
  const response = await fetch(`${BASE_URL}/api/v1/verify`, {
    method: 'POST',
    headers: {
      'Authorization': `Bearer ${API_KEY}`,
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({ email })
  });

  if (!response.ok) {
    const error = await response.json();
    throw new Error(`API Error: ${error.title} - ${error.detail}`);
  }

  return response.json();
}

/**
 * Verify multiple email addresses (Enterprise tier only)
 */
async function verifyEmailsBulk(emails) {
  const response = await fetch(`${BASE_URL}/api/v1/verify/bulk`, {
    method: 'POST',
    headers: {
      'Authorization': `Bearer ${API_KEY}`,
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({ emails })
  });

  if (!response.ok) {
    const error = await response.json();
    throw new Error(`API Error: ${error.title} - ${error.detail}`);
  }

  return response.json();
}

// Example usage
async function main() {
  try {
    // Single email verification
    console.log('Verifying single email...');
    const result = await verifyEmail('test@example.com');
    console.log('Result:', JSON.stringify(result, null, 2));

    // Check verification status
    console.log('\nVerification Status:');
    console.log(`  Email: ${result.email}`);
    console.log(`  Valid: ${result.isValid}`);
    console.log(`  Deliverable: ${result.isDeliverable}`);
    console.log(`  Disposable: ${result.isDisposable}`);
    console.log(`  Role Account: ${result.isRoleAccount}`);

    // Bulk verification example (Enterprise tier)
    console.log('\nBulk email verification...');
    const bulkResults = await verifyEmailsBulk([
      'user1@gmail.com',
      'user2@company.com',
      'fake@invalid-domain-xyz.com'
    ]);
    console.log('Bulk Results:', JSON.stringify(bulkResults, null, 2));

  } catch (error) {
    console.error('Error:', error.message);
  }
}

main();

File: examples/python/email_verification.py
"""
InstaTools API - Email Verification Example

This example demonstrates how to verify email addresses for deliverability.
"""

import os
import requests

API_KEY = os.getenv('INSTATOOLS_API_KEY', 'YOUR_API_KEY')
BASE_URL = 'https://api.instatools.com'


def verify_email(email: str) -> dict:
    """Verify a single email address."""
    response = requests.post(
        f'{BASE_URL}/api/v1/verify',
        headers={
            'Authorization': f'Bearer {API_KEY}',
            'Content-Type': 'application/json'
        },
        json={'email': email}
    )
    response.raise_for_status()
    return response.json()


def verify_emails_bulk(emails: list) -> dict:
    """Verify multiple email addresses (Enterprise tier only)."""
    response = requests.post(
        f'{BASE_URL}/api/v1/verify/bulk',
        headers={
            'Authorization': f'Bearer {API_KEY}',
            'Content-Type': 'application/json'
        },
        json={'emails': emails}
    )
    response.raise_for_status()
    return response.json()


def main():
    try:
        # Single email verification
        print('Verifying single email...')
        result = verify_email('test@example.com')
        print(f'Result: {result}')

        # Check verification status
        print('\nVerification Status:')
        print(f"  Email: {result.get('email')}")
        print(f"  Valid: {result.get('isValid')}")
        print(f"  Deliverable: {result.get('isDeliverable')}")
        print(f"  Disposable: {result.get('isDisposable')}")
        print(f"  Role Account: {result.get('isRoleAccount')}")

        # Bulk verification example (Enterprise tier)
        print('\nBulk email verification...')
        bulk_results = verify_emails_bulk([
            'user1@gmail.com',
            'user2@company.com',
            'fake@invalid-domain-xyz.com'
        ])
        print(f'Bulk Results: {bulk_results}')

    except requests.HTTPError as e:
        print(f'HTTP Error: {e.response.status_code} - {e.response.text}')
    except Exception as e:
        print(f'Error: {e}')


if __name__ == '__main__':
    main()
